### PR TITLE
Reader: Add in small post cards for related posts and features

### PR DESCRIPTION
**Pull request by blowery**
See http://calypso.localhost:3000/devdocs/app-components at the bottom for tests

### CODE REVIEW

**Comment by blowery** [comment]
screenshot

**Comment by blowery** [comment]
Softened up the corners on the images a bit 

screenshot

File: client/components/post-card/small.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	return (
+		<Card className="post-card small">
+			<a className="post-card__site-info" href={ `/read/blogs/${post.site_ID}`}>
+				<SiteIcon site={ site } size={ 16 } />
+				<span className="post-card__site-title">{ site && site.title || post.site_name }</span>
+			</a>
+			<div className="post-card__title">
+				<a className="post-card__anchor" href={ `/read/blogs/${post.site_ID}/posts/${post.ID}` }>
+				{ post.title }
+
```
Comment (TooTallNate) [inline client/components/post-card/small.jsx]: intentional empty line?

File: client/components/post-card/small.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import SiteIcon from 'components/site-icon';
+import safeImageUrl from 'lib/safe-image-url';
+import resizeImageUrl from 'lib/resize-image-url';
+
+export default function SmallPostCard( { post, site } ) {
+	return (
+		<Card className="post-card small">
+			<a className="post-card__site-info" href={ `/read/blogs/${post.site_ID}`}>
+				<SiteIcon site={ site } size={ 16 } />
+				<span className="post-card__site-title">{ site && site.title || post.site_name }</span>
```
Comment (jancavan) [inline client/components/post-card/small.jsx]: We should use a heading tag here instead of a span.

File: client/components/post-card/small.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+			</a>
+			<div className="post-card__title">
```
Comment (jancavan) [inline client/components/post-card/small.jsx]: We should use a heading tag here instead of a div.

Comment (blowery) [inline client/components/post-card/small.jsx]: Tried that. Works but devdocs is leaking styles into h2s

File: client/components/post-card/small.jsx (hunk 1 above)
Comment (blowery) [inline client/components/post-card/small.jsx]: Nope. :) Thanks.

File: client/components/post-card/small.jsx (hunk 3 above)
Comment (blowery) [inline client/components/post-card/small.jsx]: @jancavan h2? h3?

Comment (blowery) [inline client/components/post-card/small.jsx]: added an h3.

**Comment by blowery** [comment]
@jancavan @TooTallNate ready for a poke again

**Comment by folletto** [comment]
Can we check the paddings to make sure they are equalized on all the sides?

screen shot 2016-05-30 at 14 57 29

At least the image on the right should have all these three identical (`13px` for example). Same on the text to the left (`25px` for example).

**Comment by kellychoffman** [comment]
So cool to see this already built out!

I also noticed the same inconsistencies as @folletto. I adjusted featured photo and site info so they align on the 16 / 24px padding of the compact card grid.

**Comment by folletto** [comment]
Better. 

I'd even go as far as:

```
.post-card__thumbnail {
    ...
    right: 18px;
    ...
}
```

And:

```
.post-card__site-info {
    ...
    margin-top: 7px;
    ...
}

.post-card__title {
    ...
    margin: 4px 0 5px;
    ...
}
```

Which looks like:

screen shot 2016-05-30 at 15 54 58


Note: there's interference in the _devdocs_ from `.design-assets h3`. Yet another case where we should always use class to avoid CSS leaking. :)

**Comment by blowery** [comment]
Thanks @kellychoffman !

**Comment by jancavan** [comment]
Made some changes here: https://github.com/Automattic/wp-calypso/pull/5657/commits/f1b44257daff05af4b17374194aa7e338d619ddc so we don't absolutely position the image.

**Comment by folletto** [comment]
For reference, they currently look like this after the change: 

screen shot 2016-05-31 at 09 57 58

**Comment by kellychoffman** [comment]
Hmm, the vertical centering doesn't seem to add much and makes the photo look like its breaking outside the grid.

**Comment by aduth** [comment]
Seems there's quite a bit of similarity between this and our post listing for custom post types. Enough that it might be worth trying to keep them in sync, perhaps even using the same base component?

image

**Comment by blowery** [comment]
@folletto @kellychoffman @jancavan I think we should ditch the one-line title truncation and let the full title show wrapped. It looks pretty goofy right now.

Thoughts on adding in the excerpt below the title? It's not very compelling or informative yet.

**Comment by jancavan** [comment]
> I think we should ditch the one-line title truncation

Yeah, that's what I thought too.

> Thoughts on adding in the excerpt below the title?

I think in this particular case, showing just the title is fine. The titles should be enough of a "hook" to get people to click them. But if they're often bad, then displaying excerpts might help.

Btw, I think it's more semantic to make this a list-item instead of separate divs.

**Comment by blowery** [comment]
@jancavan Updated to let the title flow to multiple lines.

Noticed that the ellipsis on the site name is bumped down a few px because of the vertical-align, at least in Chrome.

Also noticed that the gap between the content and the image gets really big at some breakpoints, especially when we drop down to the single column on small screens. Since we know the image is a max 96 wide, maybe we could use calc to spec the width at 100% - 96 px?

**Comment by blowery** [comment]
Added a new test case and code for an imageless post.

Current look:

wordpress_com

**Comment by blowery** [comment]
Images look odd now. Should they line up with the top of the site title?

**Comment by kellychoffman** [comment]
> Images look odd now. Should they line up with the top of the site title?

I made a couple small changes. Now looks like:

screen shot 2016-06-02 at 11 30 22 am

**Comment by jancavan** [comment]
The layout broke in wider viewports. Fixed in: https://github.com/Automattic/wp-calypso/pull/5657/commits/51151ea39772abef59ffc2167eea4ff800b19217

**Before:**
screenshot 2016-06-02 11 50 46

**After:**
screenshot 2016-06-02 11 53 05

Thanks @kellychoffman. I think this is good to go?

**Comment by kellychoffman** [comment]
Should we turn off `:visited` styles? Make the title always gray?

**Comment by jancavan** [comment]
> Make the title always gray?

Fixed in: https://github.com/Automattic/wp-calypso/pull/5657/commits/375a46226139309f5787e8479c2ebdac0b120458

**Comment by kellychoffman** [comment]
:shipit:

**Comment by blowery** [comment]
Oooh, good catch on the :visited style 👍

**Comment by blowery** [comment]
The ellipsis on the truncated site info is still in an odd spot.

**Comment by blowery** [comment]
In the interests of completeness, I added a variant that includes the post excerpt:

wordpress_com

Thoughts?

**Comment by jancavan** [comment]
Fixed the misaligned ellipsis on site titles: https://github.com/Automattic/wp-calypso/pull/5657/commits/f54855eb5af85caa85b671a3608edb9e77fe73ab

I prefer displaying the titles only and because these cards are small and compact, it makes the entire list look busy/wordy.

@blowery What do you think about ab testing it though?

**Comment by blowery** [comment]
@jancavan that would be fine. i just wanted in the component as an option, in case some one wants a version with the excerpt present. Since this is a generic component, it should support it I think.

**Comment by blowery** [comment]
Is the gap between the site info and the post title too big? Seems big.

**Comment by jancavan** [comment]
> Is the gap between the site info and the post title too big?

I thought it was fine, but made it slightly smaller anyway.

screenshot 2016-06-03 07 56 57

**Comment by blowery** [comment]
We good? I'm happy.

**Comment by jancavan** [comment]
Yes design LGTM 👍

**Pull request merged**